pkg/kv/kvserver/replica_raft.go

```diff
@@ -1538,7 +1567,8 @@ func (r *Replica) maybeCampaignOnWakeLocked(ctx context.Context) {
 
 	leaseStatus := r.leaseStatusAtRLocked(ctx, r.store.Clock().NowAsClockTimestamp())
 	raftStatus := r.mu.internalRaftGroup.BasicStatus()
-	if shouldCampaignOnWake(leaseStatus, r.store.StoreID(), raftStatus) {
+	livenessMap, _ := r.store.livenessMap.Load().(liveness.IsLiveMap)
```

Comment:
Mind updating the comment on `store.livenessMap` to reflect the fact that this gets updated reactively (nodeIsLiveCallback) and not just on the raft tick loop?